### PR TITLE
Add validation on frontmatter author field

### PR DESCRIPTION
**Pull request by CharlRitterDev**
Add validation on frontmatter author field

### Description
Add validation on frontmatter author field

### Issue
https://spandigital.atlassian.net/browse/PRSDM-5353

### Checklist before merging

* [ ] Did you test your changes locally?
* [ ] Did you update the CHANGELOG?
* [ ] Is the documentation updated for this change? (If Required)

### CODE REVIEW

**Review by meyerhp — COMMENTED** [review]
I tested with the latest Hugo version and I couldn't see any notable performance difference.

File: layouts/partials/article/authorisempty.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{{ $value := . }}
```
[the remark this answers is not in the document]
Comment (meyerhp) [reply layouts/partials/article/authorisempty.html:1]: If checks are so fast it makes no discernible difference wether it's cached or not. I'm fine to keep the partial and see what the team thinks, imo it's better to inline it.

[the remark this answers is not in the document]
Comment (CharlRitterDev) [reply layouts/partials/article/authorisempty.html:1]: cool, happy with that, will change

**Review by Zalaras — COMMENTED** [review]
Is this not going to break current sites that currently have no author? Any spec ingestions would also be an issue...

**Comment by CharlRitterDev** [comment]
> Is this not going to break current sites that currently have no author? Any spec ingestions would also be an issue...
@Zalaras It would, but that would be part of the productionisation of the techspike

**Comment by Zalaras** [comment]
> > Is this not going to break current sites that currently have no author? Any spec ingestions would also be an issue...
> > @Zalaras It would, but that would be part of the productionisation of the techspike

So all articles would need to be updated? If so what are the plans for Apple?

**Comment by CharlRitterDev** [comment]
> > > Is this not going to break current sites that currently have no author? Any spec ingestions would also be an issue...
> > > @Zalaras It would, but that would be part of the productionisation of the techspike
> 
> So all articles would need to be updated? If so what are the plans for Apple?

@Zalaras  idk yet. This is just a techspike. That will follow.

But yeah, we want to enforce it, however we can turn off the author requirement for Apple, so it can go live without killing apple

**Review by kmorake — APPROVED** [review]

**Comment by CharlRitterDev** [comment]
Rebase went wrong, recreated with https://github.com/SPANDigital/presidium-theme-website/pull/268

**Pull request closed without merging**